### ISSUE TITLE
Improve the flexiblity of molnet function

### ISSUE THREAD

**Issue by nd-02110114** [deepchem/deepchem#1941]
Currently, The `load_xxxx` functions in `deepchem.molnet` don't have the flexiblitiy about featurizers and transforms. This sometimes leads to problems.

In my case, I couldn't use the `load_qm9` with `AdjacencyFingerprint ` , so I redefined the load_qm9 function by myself 

https://github.com/deepchem/deepchem/blob/82cb9075f16243bbb6c916f52a5ebbd5020b911d/deepchem/molnet/load_function/qm9_datasets.py#L64-L90

In addition to this, currently we can't change the argument values when instantiating `featurizer `.  I wanted to change `max_n_atoms` about `AdjacencyFingerprint` of `load_tox21`, but I couldn't. 

https://github.com/deepchem/deepchem/blob/82cb9075f16243bbb6c916f52a5ebbd5020b911d/deepchem/molnet/load_function/tox21_datasets.py#L59-L60

I think the same issues also happened about `transforms`.  Curently, many `load_xxxx` functions standardize the objective values. However, we sometimes want to normalize them, but we can't.

Therefore, I think the arguments of `load_xxxx` functions should accept `deepchem.feat.Featurizer` instances and `deepchem.feat.Transformer` instances. The following code is my thinking.

```python

"""
Tox21 dataset loader.
"""
import os
import logging
import deepchem

logger = logging.getLogger(__name__)

TOX21_URL = 'http://deepchem.io.s3-website-us-west-1.amazonaws.com/datasets/tox21.csv.gz'
DEFAULT_DIR = deepchem.utils.get_data_dir()

DEFAULT_FEATURIZERS = {
  'Raw': deepchem.feat.RawFeaturizer(),
  'ECFP': deepchem.feat.CircularFingerprint(size=1024),
  'GraphConv': deepchem.feat.ConvMolFeaturizer(),
  'AdjacencyConv': deepchem.feat.AdjacencyFingerprint(max_n_atoms=150, max_valence=6),
  'Weave': deepchem.feat.WeaveFeaturizer(),
  'smiles2img': deepchem.feat.SmilesToImage(img_size=80, img_spec='std'),
}


def load_tox21(featurizer=DEFAULT_FEATURIZERS['ECFP'],
               split='index',
               reload=True,
               K=4,
               data_dir=None,
               save_dir=None,
               **kwargs):
  """Load Tox21 datasets. Does not do train/test split"""
  # Featurize Tox21 dataset

  tox21_tasks = [
      'NR-AR', 'NR-AR-LBD', 'NR-AhR', 'NR-Aromatase', 'NR-ER', 'NR-ER-LBD',
      'NR-PPAR-gamma', 'SR-ARE', 'SR-ATAD5', 'SR-HSE', 'SR-MMP', 'SR-p53'
  ]

  if data_dir is None:
    data_dir = DEFAULT_DIR
  if save_dir is None:
    save_dir = DEFAULT_DIR

  # this is backward compatibility
  if isinstance(featurizer, str):
    featurizer = DEFAULT_FEATURIZERS[featurizer]

  if reload:
    featurizer_name = featurizer.__class__.__name__
    save_folder = os.path.join(save_dir, "tox21-featurized", str(featurizer_name))
    save_folder = os.path.join(save_folder, str(split))

    loaded, all_dataset, transformers = deepchem.utils.save.load_dataset_from_disk(
        save_folder)
    if loaded:
      return tox21_tasks, all_dataset, transformers

  dataset_file = os.path.join(data_dir, "tox21.csv.gz")
  if not os.path.exists(dataset_file):
    deepchem.utils.download_url(url=TOX21_URL, dest_dir=data_dir)

  loader = deepchem.data.CSVLoader(
      tasks=tox21_tasks, smiles_field="smiles", featurizer=featurizer)
  dataset = loader.featurize(dataset_file, shard_size=8192)
  ...
  ...
  ...
```

**Comment by rbharath** [comment]
This looks really good @nd-02110114! I've been meaning to convert the MoleculeNet loaders to work like this. As a small modification, I think you meant to also add `transformers` as an argument to the function right?

I think we should swap over all the moleculenet loaders to this style.  CC @ncfrey as well

**Comment by ncfrey** [comment]
Sounds good @nd-02110114! I will modify the loader template to accept featurizers and transformers in this way. Currently, I think `**kwargs` is set up to allow additional arguments to featurizers and splitters, but I don't know how many loaders check for kwargs to update parameters.

**Comment by nd-02110114** [comment]
>  I think you meant to also add transformers as an argument to the function right

I think this is a better situation! However, `deepchem.trans.Transformer` require `dataset` when instantiating and it is difficult to do this.

**Comment by nd-02110114** [comment]
This issue is merged in #1999
